test/test_app.py

```diff
@@ -0,0 +1,33 @@
+import pytest
+
+from flask.testing import FlaskClient
+
+from flask_calendar.app import app
+from flask_calendar.constants import SESSION_ID
+
+app.testing = True
+
+
+@pytest.fixture
+def client() -> FlaskClient:
+    return app.test_client()
+
+
+@pytest.mark.parametrize(('username', 'password', 'success'), (
+    ('', '', False),
+    ('wrong_username', '', False),
+    ('a_username', '', False),
+    ('a_username', 'wrong_password', False),
+    ('a_username', 'a_password', True)
+))
+def test_login_credentials(client: FlaskClient, username: str, password: str, success: bool) -> None:
+    rv = client.post('/do_login', data=dict(
```

Comment:
A name like `response` is way more meaningful than `rv`, don't you think?

Reply:
Sure!